src/components/Checkbox/Checkbox.vue

```diff
@@ -0,0 +1,168 @@
+<!--
+  -
+  - @author Julius Härtl <jus@bitgrid.net>
+  - @author Greta Doci <gretadoci@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+<docs>
+## One checkbox with a hint
+
+```vue
+<template>
+		<Checkbox
+			label="some text"
+			v-model="checkedTarget"/>
+</template>
+```
+</docs>
+
+<template>
+	<label>
+		<input
+			:id="id"
+			type="checkbox"
+			:checked="value"
+			:disabled="disabled"
+			@keydown.enter.exact.prevent="checkInput"
+			@change="onChange">
+		{{ label }}
+	</label>
+</template>
+
+<script>
+import GenRandomId from '../../utils/GenRandomId'
+export default {
+	name: 'Checkbox',
+	props: {
+		/**
+		 * id attribute of the checkbox element
+		 */
+		id: {
+			type: String,
+			default: () => 'checkbox-' + GenRandomId,
+
+		},
+		label: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * value of the checkbox input
+		 */
+		value: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 * disabled state of the checkbox element
+		 */
+		disabled: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	methods: {
+		checkInput(event) {
+			// by clicking we also trigger the change event
+			this.$refs.label.click()
+		},
+		onChange(event) {
+			/**
+			 * Emitted when the checkbox state is changed
+			 * @type {boolean}
+			 */
+			this.$emit('update:checked', this.$refs.checkbox.checked)
```

Comment:
If you want to support `v-model`, like you gave an example of above, you will have to emit an event called `update:value`.
```suggestion
			this.$emit('update:value', this.$refs.checkbox.checked)
```

Reply:
`update:value` is not for v-model. But for the sync modifier.
v-model will rely on change/input depending on the elemtn type :wink:

Reply:
True, my bad, sorry. But since there is no prop called `checked` it should be renamed anyway